src/main.rs

```diff
@@ -78,7 +67,7 @@ fn cache_visit(cache: Store) -> impl warp::Reply {
         // base case: cache still valid, only need a read lock.
         // should be very fast
         let timed_string = cache.read().unwrap();
-        if Instant::now() < timed_string.time {
+        if timed_string.is_valid() {
```

Comment:
moved this to a function, could also be something like `!timed_string.is_expired()` but using the negative seemed unnecessary.